include/pybind11/pybind11.h

```diff
@@ -1407,12 +1407,13 @@ template <typename Type> class enum_ : public class_<Type> {
             !std::is_same<detail::first_of_t<arithmetic_tag, void, Extra...>,
                           void>::value;
 
-        auto entries = new std::unordered_map<Scalar, const char *>();
+        auto entries = new pybind11::dict();
```

Comment:
use of ``new`` here is problematic (this will never be cleaned up.) Better to use standard python objects and rely on reference counting.